src/gui/Properties.cpp

```diff
@@ -40,3 +40,10 @@ void Properties::bindObject(const int objectId) {
                        "<font color='$Color-PropertiesObjectTypeText'>"+breakStringAtCaps(objectType)+"</font>)";
     nameWidget->setText(Globals::theme->process(nameType));
 }
+
+void Properties::rewriteObjectNameAndType() {
```

Comment:
Only (minor) critique is the name of this function is a little awkward.  Typically, this sort of operation would be called a "redraw" or even a "draw" routine.  Calling it a "rewrite" sounds a bit like the name is being changed (e.g., rewriting a file).